interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -1533,6 +1534,20 @@ def test_get_state(self):
         self.assertEqual(P.shape, (4,))
         self.assertEqual(Y.shape, (4, self.gas.n_species))
 
+    @unittest.expectedFailure
+    def test_idealgas_getters(self):
```

Comment:
This ensures that all getters can actually be used, i.e.
```python
        getters = 'TDPUVHSXY'
        all_getters = [k for k in states.__dir__()
                       if not set(k) - set(getters) and len(k)>1]
```
this would not be true for a setter `TQ`, which is part of `states__dir__()`, but will fail with an attribute error. I added comments.

Reply:
OK, that makes sense, but I don't think that's what this test achieves. What will happen here is that the first property that works will not raise an `AttributeError` and then test will fail. If you want to test all the getters that should work, you could build that list using the properties of the `ThermoPhase` class. Then you can invert the structure of the test by just eliminating `with` clause.

I think that we should reserve `expectedFailure` for tests that should pass but do not because of a known bug. The only cases where we currently use `expectedFailure` are for some of the equilibrium tests, for which I believe there are open issues in the tracker (although I realize there is unfortunately no comment linking to the relevant issue).

I would also suggest using `dir(foo)` over `foo.__dir__()` and `getattr(foo, bar)` over `foo.__getattribute__(bar)`.

Reply:
> I think that we should reserve `expectedFailure` for tests that should pass but do not because of a known bug.

Ok. I changed the tests accordingly. 

> What will happen here is that the first property that works will not raise an `AttributeError` and then test will fail. 

Indeed! Thanks for pointing that out - on the upside, the unit test surfaced an old typo (`self.p` instead of `self.P` in `PureFluid.PV`).